### PR TITLE
Complete core_task_5

### PR DESCRIPTION
**Pull request by LiamCurran95**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -101,12 +102,12 @@ describe("/api/articles endpoint", () => {
 					expect(msg).toBe("Bad request.");
 				});
 		});
-		test("Status 404 - Valid ID - Doesn't exist within database.", () => {
+		test("Status 404 - Valid ID - Article doesn't exist within database.", () => {
 			return request(app)
 				.get(`/api/articles/234`)
 				.expect(404)
 				.then(({ body: { msg } }) => {
```
Comment (DCrawley94) [inline __tests__/app.test.js:109]: Nice clear test descriptions!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -117,6 +118,15 @@ describe("/api/articles endpoint", () => {
 					expect(msg).toBe("Path not found.");
 				});
 		});
+		test("Status 404 - Valid ID - Article has no comments (comment_count)", () => {
```
Comment (DCrawley94) [inline __tests__/app.test.js:121]: I'd argue that this should be a 200. Even if there's no comments on an article a user would still want to read it! 

If an article can't be read until it has comments it'll never be read!

File: controllers/articles.controllers.js [hunk 3]
```diff
@@ -15,7 +16,11 @@ exports.getArticles = (req, res, next) => {
 
 exports.getArticlesById = (req, res, next) => {
 	const { article_id } = req.params;
-	Promise.all([fetchArticlesById(article_id), checkArticleExists(article_id)])
+	Promise.all([
+		fetchArticlesById(article_id),
+		checkArticleExists(article_id),
```
Comment (DCrawley94) [inline controllers/articles.controllers.js:21]: We shouldn't need to check if the article exists in this case:
If the ID is valid but doesn't exist then the length check in the model will deal with it
If the ID invalid then PSQL will throw an error which we can deal with in an error handler

Also see above about what to do if an article has no comments :)

**Review by DCrawley94 — CHANGES_REQUESTED** [review]
I think you've got slightly confused about what we want to do in regards to the comment_count - there's nothing wrong with an article having a comment_count of 0 :)

**Review by Poonam-raj — APPROVED** [review]
Just that comment on extracting the article from the array so you're just returning the object on a key of articles - otherwise looks great

**Pull request closed without merging**